### PR TITLE
Initial version of admission control in RntbdServiceEndpoint

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java [hunk 1]
```diff
@@ -396,6 +406,11 @@ public int maxRequestsPerChannel() {
             return this.maxRequestsPerChannel;
         }
 
+        public int maxConcurrentRequestsPerEndpoint()
+        {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java]: code style `{` on the same line as method name

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java [hunk 2]
```diff
@@ -582,6 +599,11 @@ public Builder(ConnectionPolicy connectionPolicy) {
                 this.maxBufferCapacity = DEFAULT_OPTIONS.maxBufferCapacity;
                 this.maxChannelsPerEndpoint = connectionPolicy.getMaxConnectionsPerEndpoint();
                 this.maxRequestsPerChannel = connectionPolicy.getMaxRequestsPerConnection();
+
+                this.maxConcurrentRequestsPerEndpoint = Math.max(
+                    10000,
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java]: should we put in the DEFAULT_OPTIONS?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/AsyncRntbdRequestRecord.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public final class AsyncRntbdRequestRecord extends RntbdRequestRecord {
+
+    private final RntbdRequestTimer timer;
+
+    public AsyncRntbdRequestRecord(final RntbdRequestArgs args, final RntbdRequestTimer timer) {
+
+        super(args);
+
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/AsyncRntbdRequestRecord.java]: new line remove.

**Review by moderakh — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/AsyncRntbdRequestRecord.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/AsyncRntbdRequestRecord.java]: Fixed

**Pull request merged**